WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -318,44 +340,68 @@ public void onResponse(JSONObject response) {
                         mRemoteJpSettings.ssoActive = sentJpData.ssoActive;
                         mRemoteJpSettings.ssoMatchEmail = sentJpData.ssoMatchEmail;
                         mRemoteJpSettings.ssoRequireTwoFactor = sentJpData.ssoRequireTwoFactor;
-                        notifySavedOnUiThread(null);
+                        onSaveResponseReceived(null);
                     }
                 }, new RestRequest.ErrorListener() {
                     @Override
                     public void onErrorResponse(VolleyError error) {
                         AppLog.w(AppLog.T.API, "Error updating Jetpack settings: " + error);
-                        notifySavedOnUiThread(error);
+                        onSaveResponseReceived(error);
                     }
                 });
     }
 
     private void pushJetpackMonitorSettings() {
-        final Map<String, String> params = serializeJetpackMonitorParams();
-        if (params.isEmpty()) {
-            return;
-        }
-
         // The response object doesn't contain any relevant info so we have to create a copy of values
         // being sent over the network in case mJpSettings is modified while awaiting response
         final JetpackSettingsModel sentJpData = new JetpackSettingsModel(mJpSettings);
+        ++mSaveRequestCount;
         WordPress.getRestClientUtilsV1_1().setJetpackMonitorSettings(
-                mSite.getSiteId(), params, new RestRequest.Listener() {
+                mSite.getSiteId(), serializeJetpackMonitorParams(), new RestRequest.Listener() {
                     @Override
                     public void onResponse(JSONObject response) {
                         AppLog.d(AppLog.T.API, "Jetpack Monitor module updated");
                         mRemoteJpSettings.emailNotifications = sentJpData.emailNotifications;
                         mRemoteJpSettings.wpNotifications = sentJpData.wpNotifications;
-                        notifySavedOnUiThread(null);
+                        onSaveResponseReceived(null);
                     }
                 }, new RestRequest.ErrorListener() {
                     @Override
                     public void onErrorResponse(VolleyError error) {
                         AppLog.w(AppLog.T.API, "Error updating Jetpack Monitor module: " + error);
-                        notifySavedOnUiThread(error);
+                        onSaveResponseReceived(error);
                     }
                 });
     }
 
+    private void onFetchResponseReceived(Exception error) {
+        if (error != null) {
+            mWasFetchError = true;
+            mFetchError = error;
+        } else {
+            notifyUpdatedOnUiThread();
```

Comment:
We previously had this call for when an error occurs as well. Any reason we don't need it any more?